abis_mapping/base/mapper.py

```diff
@@ -66,6 +66,76 @@ def apply_mapping(
             rdflib.Graph: ABIS Conformant RDF Sub-Graph from Raw Data Chunk.
         """
 
+    @final
+    @classmethod
+    def extract_extra_fields(
+        cls,
+        row: frictionless.Row
+    ) -> dict[str, Any]:
+        """Extracts extra values from a row not in template schema.
+
+        Args:
+            row (frictionless.Row): Row of data including extra rows.
+
+        Returns:
+            dict[str, Any]: Dictionary containing extra values, if any.
+        """
+        # Get schema consisting of extra fields
+        extra_schema = cls.extra_fields_schema(row)
+
+        # Create dictionary consisting row data from extra fields only
+        return {field: row[field] for field in extra_schema.field_names if row[field] is not None}
+
+    @final
+    @classmethod
+    def extra_fields_schema(
+        cls,
+        data: types.ReadableType | frictionless.Row
+    ) -> frictionless.Schema:
+        """Creates a schema with all extra fields found in data.
+
+        Args:
+            data (types.ReadableType | frictionless.Row): Readable raw csv
+                data or frictionless row, expected to contain more columns
+                than included in the template's schema.
+
+        Returns:
+            frictionless.Schema: A schema object, the fields of which are only
+                the extra fields not a part of a template's official schema.
+        """
+        # Construct official schema
+        existing_schema: frictionless.Schema = frictionless.Schema.from_descriptor(cls.schema())
+
+        # Handle for different data types
+        if isinstance(data, frictionless.Row):
+            # Get set of fieldnames of row
+            actual_fieldnames = set(data.field_names)
+
+            # Create schema from row fields
+            actual_schema = frictionless.Schema(fields=data.fields)
+
+        else:
+            # Construct resource from data and infer
+            resource = frictionless.Resource(
```

Comment:
I originally had it planned in my mind to find the extra columns before iterating over rows, but then saw that would come with potentially more complexity when it came time to implement it in the apply_mapping method. I then implemented the row-wise technique. I'm not sure which method will be more optimised but if the mapping becomes the bottleneck this could be something to target potentially. 

I have removed the readableType method overloading.